src/UserAction/ExecutorFactory.php

```diff
@@ -172,19 +171,7 @@ protected function createExecutor(UserAction $action, View $owner, string $requi
             }
         }
 
-        $executor = Factory::factory($seed);
-        if ($executor instanceof Modal) {
-            // add modal to app->html for proper rendering on callback.
-            if (!isset($owner->getApp()->html->elements[$executor->short_name])) {
-                // very dirty hack, @TODO, attach modals in the standard render tree
-                // but only render the result to a different place/html DOM
-                $executor->viewForUrl = $owner;
-                $executor = $owner->getApp()->html->add($executor, 'Modals'); //->setAction($action);
-            }
-        } else {
-            $executor = $owner->add($executor);
-        }
-
+        $executor = $owner->add(Factory::factory($seed));
```

Comment:
We make Modal standard part of render tree, this is the trick?

If yes, well done!

Reply:
Yes, but since Fomantic will move the modal HTML elsewhere in the document, Modal has to be registered within App in order for it to re-render them for JSON output.

Reply:
👍